tests/test_up_to_date_pypi.py

```diff
@@ -547,3 +547,130 @@ def side_effect(url: str, timeout):
             "Major upgrades:\ndummy_package_major not at latest pypi version: 2.0.0, is at: 1.0.0\n\nMinor upgrades:\ndummy_package_minor not at latest pypi version: 1.1.0, is at: 1.0.0\n\nPatch upgrades:\ndummy_package_patch not at latest pypi version: 1.0.1, is at: 1.0.0"
             in system_print
         )
+
+
+
+@pytest.mark.parametrize(
+    "release_file_content, ignore_argument",
+    [
+        pytest.param(
+            """
+                    dummy_package_patch: 1.0.0
+                    dummy_package_minor: 1.0.0
+                    dummy_package_major: 1.0.0
+                    dummy_package_should_not_update: main
+                """,
+            "main",
+            id="ignore_main_version",
+        ),
+        pytest.param(
+            """
+                    dummy_package_patch: 1.0.0
+                    dummy_package_minor: 1.0.0
+                    dummy_package_major: 1.0.0
+                    dummy_package_should_not_update: 1.0.0
+                """,
+            "should_not_update",
+            id="ignore_semantic_version",
+        ),
+    ],
+)
+def test_upgrade_ignore_flag(
+    release_file_content, ignore_argument, tmpdir, monkeypatch, capsys
+):
+    repository_file_content = """
+    dummy_package_patch:
+      1.0.0:
+        source: pypi
+        make: pip
+        maintainer: scout
+
+    dummy_package_minor:
+      1.0.0:
+        source: pypi
+        make: pip
+        maintainer: scout
+
+    dummy_package_major:
+      1.0.0:
+        source: pypi
+        make: pip
+        maintainer: scout
+
+    dummy_package_should_not_update:
+      1.0.0:
+        source: pypi
+        make: pip
+        maintainer: scout
+
+      main:
+        source: pypi
+        make: pip
+        maintainer: scout
+    """
+
+    with tmpdir.as_cwd():
+        with open("repository_file.yml", "w") as fout:
```

Comment:
encoding="utf-8"...

Reply:
Personally I prefer 

```
from pathlib import Path

Path("repository_file.yml").write_text(repository_file_contents, encoding="utf-8")
```

always over  `with open()`

See e.g. https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/

Reply:
I might start using pathlib for working with paths, but it does not seem like such a big difference for writing or reading a file in the current directory.